gh-99240: Fix double-free bug in Argument Clinic str_converter generated code (#99241)
python/cpython: Misc/NEWS.d/next/Library/2022-11-08-15-54-43.gh-issue-99240.MhYwcz.rst

> Add "post_parsing" section to Argument Clinic generated code.
> Fix double-free bug in Argument Clinic str_converter generated code by
> moving memory clean up out of "exit" label.

Reviewer: How about this?

```suggestion
Fix double-free bug in Argument Clinic ``str_converter`` by
extracting memory clean up to a new ``post_parsing`` section.
```
